src/dataset_impl.h

```diff
@@ -162,6 +163,20 @@ class DatasetImpl : public Dataset {
         return nullptr;
     }
 
+    DatasetPtr
+    Paths(const std::string* paths) override {
+        this->data_[DATASET_PATHS] = paths;
+        return shared_from_this();
+    }
+
+    const std::string*
+    GetPaths() const override {
```

Comment:
Add memory free logic for new `path_` object

Reply:
done